### PR TITLE
[sd-stub] add support for embedding devicetree

### PR DESCRIPTION
**Pull request by mxre**
Inspired by https://github.com/systemd/systemd/pull/20601#issuecomment-925680849 and based on https://github.com/systemd/systemd/pull/19417

This PR adds support for a new section in the sd-stub ".fdt" to embed a compiled devicetree (DT). 

This does not solve the problem mention here: https://github.com/systemd/systemd/pull/20601#issuecomment-925724947 since this PR only allows for one DT to be embedded.

### CODE REVIEW

File: src/boot/efi/devicetree.c [hunk 1]
```diff
@@ -109,6 +109,31 @@ EFI_STATUS devicetree_install(struct devicetree_state *state,
         return uefi_call_wrapper(BS->InstallConfigurationTable, 2, &EfiDtbTableGuid, PHYSICAL_ADDRESS_TO_POINTER(state->addr));
 }
 
+EFI_STATUS devicetree_install_from_memory(struct devicetree_state *state,
+                VOID *dtb_buffer, UINTN dtb_length) {
```
Comment (poettering) [inline src/boot/efi/devicetree.c]: const VOID*, no?

File: src/boot/efi/stub.c [hunk 2]
```diff
@@ -149,12 +151,15 @@ EFI_STATUS efi_main(EFI_HANDLE image, EFI_SYSTEM_TABLE *sys_table) {
                 [SECTION_LINUX]   = (const CHAR8*) ".linux",
                 [SECTION_INITRD]  = (const CHAR8*) ".initrd",
                 [SECTION_SPLASH]  = (const CHAR8*) ".splash",
+                [SECTION_FDT]     = (const CHAR8*) ".fdt",
```
Comment (medhefgo) [inline src/boot/efi/stub.c]: Not an expert, but wouldn't naming the section ".dtb" make more sense (I assume it must be in binary form)?

Comment (mxre) [reply src/boot/efi/stub.c]: dtb is probably a better name, it keeps the name in line with kernel parameters.

I was just really unsure. dtb is what the file format (and suffix) is called but the whole device tree spec is call fdt (flat device tree)

Comment (poettering) [reply src/boot/efi/stub.c]: Maybe call it .devicetree?

Comment (medhefgo) [reply src/boot/efi/stub.c]: Too long

**Comment by poettering** [comment]
The place to document this is in systemd-stub(7), i.e. man/systemd-stub.xml

**Pull request merged**